File: src/main/java/com/kevin/student/Student.java
package com.kevin.student;

public class Student {

    private String id;
    private String name;
    private int english;
    private int math;

    public Student(String id, String name, int english, int math) {
        this.id = id;
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public void print(){
        System.out.println(name + "\t" + english + "\t" + math + "\t" +
                getAverage() + "\t" + (getAverage()>=60 ?"Pass":"Failed"));

        char grading  = 'F';

        switch (getAverage()/10){
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;

            default:
                grading = 'F';
        }

        System.out.println("\t" + grading);



//        if(getAverage() >= 60 ){
//            System.out.println("\tPASS");
//
//        }else{
//            System.out.println("Fail");
//        }
    }

    public int highest(){
        return Math.max(english, math);
    }

    public int getAverage(){
        return (english + math ) / 2 ;
    }




}

File: src/main/kotlin/com/kevin/student/Student.kt
package com.kevin.student

import java.util.*

fun main(args: Array<String>) {
    //userInput()

    val stu = Student1("Gery",100,70)
    stu.print()
    stu.nameCheck()

    println("High Score ${stu.highest()}")

}

fun userInput() {
    val scanner = Scanner(System.`in`)

    println("Please enter student's name:")
    //val name = scanner.next();
    val name = null

    println("Please enter english score:")
    val english = scanner.nextInt()

    println("Please enter math score:")
    val math = scanner.nextInt()
}

class Student1(var name :String? , var english : Int , var math :Int){

    fun print(){
        println("$name \t $english \t $math \t ${getAverage()} \t ${passOrFailed()}")

        println("\t" + grading())
    }

    fun passOrFailed() = if(getAverage()>=60) "Pass" else "Failed"

    fun nameCheck(){
        println(name?.length)
    }

    fun highest () =if(english > math){
        println("English")
        english
    }else{
        println("Math")
        math
    }

    fun getAverage() =(english+math)/2

    fun grading() =when(getAverage()){
        in 90..100 ->'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else ->'F'
    }

}